### DISCUSSION TITLE
Provide Exemplary Case Studies

### DISCUSSION THREAD

**Discussion by Green-Software** [Green-Software-Foundation/sci#196] [SCI feedback]
After we release the alpha version of SCI, the community may still need more guidelines on how to apply it to their software products. 

We may work together to come up with several exemplary case studies to help the community better understand how to leverage SCI to reduce the carbon emissions of their software.  

We can also consider publish the updated version of SCI metrics with these case studies at top journals in the field such as IEEE Transactions on Sustainable Computing or Journal of Sustainable Computing, which will attract attention from a broader audience.

**Comment by Henry-WattTime** [comment]
I think this is an excellent idea, but I think should be part of the followon efforts. I think we should publish this Alpha and then start building case studies. @buchananwp I think the work you've done might be a perfect example.

**Comment by jawache** [comment]
Agreed, I think @Green-Software was talking about doing this post Alpha release anyway. We'd briefly touched on this at the end of the last Standard WG call, focussing the next phase on applying the SCI to some use cases. 

I think we should take a few baby steps first, perhaps 3/4 internal case studies with **very simple** apps from a broad set of categories to demonstrate it can be applied in a variety of contexts. 

- A monolith desktop windows app
- A very simple cloud workload
- A popular open-source library (numpy? tensorflow?)
- Our own website.

The equation looks simple but getting the data is incredibly challenging, it's going to be a lot harder to calculate than we think!

**Reply by ArneTR** [reply]
Hey Asim,

I am picking up this rather old post in your discussions tab as it hast the best fit to my question and I believe it is still an open issue.

Currently I help at Green Coding Berlin (https://www.green-coding.org/) and am working on a software certification tool that in essence is able to measure the resource usage of arbitrary software.

The software does this by consuming a Public Github repository, where it exepcts a JSON file that contains relevant information on how to launch and setup the relevant docker containers and the run a "typical usage flow" on the software (https://github.com/green-coding-berlin/green-metric-demo-software/blob/main/usage_scenario.json).

The mechanism is very similar to https://greenframe.io/ which you maybe know, as it is also listed in youf [GSF Awesome Software repo](https://github.com/Green-Software-Foundation/awesome-green-software)

The difference is, that we build it to support command-line, webservice and desktop applications as well and that is does not have the licensing issues of Greenframe (which has no license and this is not free use)

The current stage is a first working prototype that can run command line applications. In the repository for the demo software a Python hash algorithm and a ML (Sklearn) example is given: https://github.com/green-coding-berlin/green-metric-demo-software

The measurement application itself is also available on Github (https://github.com/green-coding-berlin/green-metrics-tool). 
A running demo, where you can submit a software can be found here: https://green-metric.codetactics.de/request.html .
Since you have to supply a repo with a correct JSON file it is the easisest to supply our demo repository with the command line application here.
Even easier is to look at the generated report: https://green-metric.codetactics.de/?id=04ca9a03-a6a0-4fa5-ac84-2e0a2a5a1375

The current implementation focuses on the resource consumptions on a CPU / Memory / Network level but could be easily extended to show the SCI metrics. The relevant parameters would be given either through self-discovery or via a config file / user input.

Is this project of any interest for GSF?

The current roadmap for the tool is to allow a software certification according to the https://www.blauer-engel.de/en/productworld/resources-and-energy-efficient-software-products .

But it would also be an honor to have the tool somehow being useful or even integrated into the GSF.
It could hopefully help to make the SCI standard more approachable and easier to apply to running projects.
Given the suggestions in your post it could be very easily be adapted to measure your website (through a docker container running chrome and a Selenium style usage_scenarion.json that mimics a typical user interaction with the website recording network traffic, CPU and memory) the numpy example etc.

I read through all the discussions issues and concluded that this was the best spot to post my questions. I hope I have not missed that you maybe even have such an implementation, but I could not view many resources as I did not have any access (Ex.: The report in Google Docs in https://github.com/Green-Software-Foundation/software_carbon_intensity/issues/227) 

Looking forward to your reply.

Best,

Arne

**Reply by atg-abhishek** [reply]
Hi @ArneTR thank you so much for sharing this - very interesting approach and would be great to explore how this can be extended to incorporate SCI metrics. @seanmcilroy29 can we add this as an agenda item for discussion in the Standards WG?

I think this might even be a great thing to include with the Innovation WG? @dtoakley @pierlag cc @jawache

**Comment by will-iamalpine** [comment]
That sounds great! Especially since 'E' and 'M' are readily available :)

Best,
Will

**Comment by navveenb** [comment]
We had created a reference document on how to apply SCI to real-world use case. Pls check. Details are posted in this discussion -  https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/120.

**Reply by atg-abhishek** [reply]
I've left a comment there to convert this into a wiki entry so that we can keep everything in the same set of tools that we're using here

**Comment by atg-abhishek** [comment]
We could have a base set of case studies that are applied to each version of the SCI release with the intention of illustrating the changes in the SCI over time - it would not be used as a gating mechanism

**Comment by atg-abhishek** [comment]
1. Case study on UNIX (~to the Windows one proposed by @jawache ) 
2. Some minimal version of an SRE case study @greenhsu123  (some reduction in the complexity can drive efficiency as well - @dubrie ) 
3. If there is an SLO in place, you can shift the location and timing 
4. @atg-abhishek to link back the existing SRE discussion

**Comment by srini1978** [comment]
> 1. Case study on UNIX (~to the Windows one proposed by @jawache )
> 2. Some minimal version of an SRE case study @greenhsu123  (some reduction in the complexity can drive efficiency as well - @dubrie )
> 3. If there is an SLO in place, you can shift the location and timing
> 4. @atg-abhishek to link back the existing SRE discussion

@atg-abhishek  @greenhsu123 We can also pick up the discussion we had in #110
